pkg/pdp/aggregator/aggregate_buffer.go

```diff
@@ -0,0 +1,143 @@
+package aggregator
+
+import (
+	"context"
+	// for go:embed
+	_ "embed"
+	"fmt"
+	"sync"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/namespace"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/datamodel"
+	"github.com/ipld/go-ipld-prime/schema"
+	"github.com/storacha/go-libstoracha/capabilities/types"
+	"github.com/storacha/go-libstoracha/ipnipublisher/store"
+	"go.uber.org/fx"
+
+	"github.com/storacha/piri/internal/ipldstore"
+)
+
+//go:embed aggregate_buffer.ipldsch
+var bufferSchema []byte
+
+var bufferTS *schema.TypeSystem
+
+func init() {
+	ts, err := ipld.LoadSchemaBytes(bufferSchema)
+	if err != nil {
+		panic(fmt.Errorf("loading submission buffer schema: %w", err))
+	}
+	bufferTS = ts
+}
+
+type Aggregates struct {
+	Pending []datamodel.Link
```

Comment:
If they exist here they are always pending right?
```suggestion
	Roots []datamodel.Link
```

```suggestion
	Pieces []datamodel.Link
```

Reply:
But an "Aggregates" doesn't have a list of "Pending" roots/pieces. They're not pending with respect to a collection they happen to be in, they're pending with respect to a larger operation. So, is this struct actually the state of that operation? Should it be named after the operation?